tests/src/test/scala/whisk/core/controller/test/ActivationsApiTests.scala

```diff
@@ -526,18 +526,33 @@ class ActivationsApiTests extends ControllerTestCommon with WhiskActivationsApi
     implicit val materializer = ActorMaterializer()
     val activationStore = SpiLoader
       .get[ArtifactStoreProvider]
-      .makeStore[WhiskEntity](whiskConfig, _.dbActivations)(
-        classTag[WhiskEntity],
-        WhiskEntityJsonFormat,
+      .makeStore[WhiskActivation]()(
+        classTag[WhiskActivation],
+        WhiskActivation.serdes,
         WhiskDocumentReader,
         system,
         logging,
         materializer)
     implicit val tid = transid()
-    val entity = BadEntity(namespace, EntityName(ActivationId().toString))
-    put(activationStore, entity)
 
-    Get(s"$collectionPath/${entity.name}") ~> Route.seal(routes(creds)) ~> check {
+    class BadActivation(override val namespace: EntityPath,
```

Comment:
Why are these changes needed in this context?

Reply:
The test here was validating that deserialization issue triggers 500 response. For this it injected a WhiskEntity in Activation table by "misusing" the dbName parameter

```
 val activationStore = SpiLoader
      .get[ArtifactStoreProvider]
      .makeStore[WhiskEntity](whiskConfig, _.dbActivations)(
        WhiskEntityJsonFormat,
```
Now as the db name is now mapped implicitly by entity type class this would not work. So test has been modified to use a `BadActivation` which emits wrong json and thus triggers the deserialization fault

Reply:
Ah got it, could you add a comment to the `toJson` method stating that it breaks the deserialization by removing the subject entry?